packages/apollo-links/src/authHttpLink.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020-2022 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { from, ApolloLink, HttpOptions } from '@apollo/client/core';
+
+import { AuthLink } from './auth-link';
+import { DynamicHttpLink } from './dynamicHttpLink';
+import agreementMananger from './agreementMananger';
+import { creatErrorLink } from './errorLink';
+import { retryLink } from './retryLink';
+import { Logger } from "./types";
+
+interface AuthHttpOptions {
+  authUrl: string;          // auth service url
+  projectChainId: string;   // genesis hash of the chain
+  httpOptions: HttpOptions; // http options for init `HttpLink`
+  deploymentId: string;     // deployment id of the project
+  logger: Logger       // logger for `AuthLink`
+  backupDictionary?: string; // backup dictionary for `AuthLink`
```

Comment:
We should name it more generically because this apollo link isn't limited to dictionary queries.